File: examples/exp-ticket-support/functions/human_escalate.py
import json


def new():
    return Function()


class Function:
    async def handle(self, scope, receive, send):
        message = await receive()
        body = message.get("body", b"")
        event = json.loads(body.decode("utf-8")) if body.strip() else {}
        risk = int(event.get("risk", 0))

        result = {
            **event,
            "action": "human-escalate",
            "priority": "p1" if risk >= 90 else "p2",
            "queue": "vip-support" if event.get("user_level") == "vip" else "support",
            "handled_by": scope["minik8s"]["name"],
        }
        await respond(send, result)


async def respond(send, result):
    payload = json.dumps(result).encode("utf-8")
    await send({
        "type": "http.response.start",
        "status": 200,
        "headers": [[b"content-type", b"application/json"]],
    })
    await send({"type": "http.response.body", "body": payload})

File: examples/exp-ticket-support/functions/human_escalate.rs
use crate::FunctionContext;
use serde_json::{Map, Value};

pub fn handle(event: Value, ctx: &FunctionContext) -> Value {
    let risk = int_field(&event, "risk");
    let user_level = string_field(&event, "user_level", "");
    let mut result = object_or_empty(event);
    result.insert(
        "action".to_string(),
        Value::String("human-escalate".to_string()),
    );
    result.insert(
        "priority".to_string(),
        Value::String(if risk >= 90 { "p1" } else { "p2" }.to_string()),
    );
    result.insert(
        "queue".to_string(),
        Value::String(
            if user_level == "vip" {
                "vip-support"
            } else {
                "support"
            }
            .to_string(),
        ),
    );
    result.insert("handled_by".to_string(), Value::String(ctx.name.clone()));
    Value::Object(result)
}

fn string_field(value: &Value, field: &str, default: &str) -> String {
    value
        .get(field)
        .and_then(Value::as_str)
        .unwrap_or(default)
        .to_string()
}

fn int_field(value: &Value, field: &str) -> i64 {
    match value.get(field) {
        Some(Value::Number(number)) => number.as_i64().unwrap_or(0),
        Some(Value::String(text)) => text.parse::<i64>().unwrap_or(0),
        _ => 0,
    }
}

fn object_or_empty(value: Value) -> Map<String, Value> {
    match value {
        Value::Object(object) => object,
        _ => Map::new(),
    }
}
